pkg/trace/obfuscate/sql.go

```diff
@@ -153,35 +153,23 @@ func (f *groupingFilter) Reset() {
 // function is generic and the behavior changes according to chosen tokenFilter implementations.
 // The process calls all filters inside the []tokenFilter.
 func (o *Obfuscator) obfuscateSQLString(in string) (string, error) {
-	ignoreEscapes := o.ShouldIgnoreEscapes()
-	tokenizer := NewSQLTokenizer(in, ignoreEscapes)
+	literalEscapes := o.LiteralEscapes()
+	tokenizer := NewSQLTokenizer(in, literalEscapes)
 	filters := []tokenFilter{&discardFilter{}, &replaceFilter{}, &groupingFilter{}}
 
 	out, err := attemptObfuscation(tokenizer, filters)
-	if err != nil {
-		// If the tokenizer failed, but saw an escape character in the process
-		// retry with opposite setting.
-		if tokenizer.HasSeenEscape() {
-			tokenizer = NewSQLTokenizer(in, !ignoreEscapes)
-			var err2 error
-			out, err2 = attemptObfuscation(tokenizer, filters)
-			if err2 == nil {
-				// If the second attempt succeeded, make the config change
-				err = nil
-				o.SetIgnoreEscapes(!ignoreEscapes)
-			}
+	if err != nil && tokenizer.SeenEscape() {
+		// If the tokenizer failed, but saw an escape character in the process,
+		// try again treating escapes differently
+		tokenizer = NewSQLTokenizer(in, !literalEscapes)
+		var err2 error
```

Comment:
This `var` is weird here. I think we can drop it. Also, you're missing the case when `err2 != nil`. Maybe:
```go
out, err2 := attemptObfuscation(tokenizer, filters)
if err2 == nil {
    o.SetLiteralEscapes(!literalEscapes)
    return out, nil
}
return out, err
```
I think this is more idiomatic and possibly easier to reason about.

Reply:
When err2 != nil we want to return the original error.

Reply:
Yep, this is what happens in my example. Unless I’m misunderstanding something. If the new error (err2) is nil we return the result, otherwise we return the original error.